pinot-common/src/main/java/org/apache/pinot/common/data/Segment.java

```diff
@@ -18,14 +18,14 @@
  */
 package org.apache.pinot.common.data;
 
-import javax.annotation.Nonnull;
+import org.jspecify.annotations.NonNull;
```

Comment:
We are trying to follow the convention of only annotating the `nullable` fields, and treat the fields not annotated as non-null. Since you are fixing all annotations, let's remove the `Nonnull` ones.